### PR TITLE
feat(cloudtrail): create cloudwatch event without needing to create a Trail

### LINKED ISSUE #6716: cloudtrail: Also make onCloudTrailEvent a static method
### The Question

Can you add an import/lookup method to the trail construct?

I have been tasked with multiple projects to fire to lambda based on cloud trail events. Each time I have had to create a new trail because thre is no lookup method. Is there currently a work around? We have already reached out hard cap of 5 trails.  
  
Thanks

### Environment
  - **CDK CLI Version:** 1.26.0
  - **Module Version:** 1.26.0
  - **OS:** all
  - **Language:** all

### Other information
AWS has a hard limit of 5 cloud trails per account. Not having a method to import existing trails on the trail construct really limits what you can do with this service in CDK.

### PR DESCRIPTION
**Pull request by nija-at**
closes #6716

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudtrail/README.md [hunk 1]
```diff
@@ -92,4 +92,33 @@ trail.addLambdaEventSelector(["arn:aws:lambda"]);
 
 // Add an event selector to log data events for the provided Lambda functions.
 trail.addLambdaEventSelector([lambdaFunction.functionArn]);
+```
+
+Using the `Trail.onEvent()` API, a Cloudwatch event rule can be created that gets triggered for
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-cloudtrail/README.md:97]:
```suggestion
Using the `Trail.onEvent()` API, an EventBridge rule can be created that gets triggered for
```

might be for a different PR
motivation: consistency with upcoming change in #8020

File: packages/@aws-cdk/aws-cloudtrail/README.md [hunk 2]
```diff
@@ -92,4 +92,33 @@ trail.addLambdaEventSelector(["arn:aws:lambda"]);
...
+every event logged in CloudTrail.
+To only use the events that are of interest, either from a particular service, specific account or
+time range, they can be filtered down using the APIs available in `aws-events`. The following code
+filters events for S3 and from a specific AWS account and triggers a lambda function. See [Events Delivered via
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-cloudtrail/README.md:101]:
```suggestion
filters events for S3 from a specific AWS account and triggers a lambda function. See [Events delivered via
```

alternatively if we want to be consistent with the docs that it points to, also capitalize `Via`

File: packages/@aws-cdk/aws-cloudtrail/README.md [hunk 3]
```diff
@@ -92,4 +92,33 @@ trail.addLambdaEventSelector(["arn:aws:lambda"]);
 
 // Add an event selector to log data events for the provided Lambda functions.
 trail.addLambdaEventSelector([lambdaFunction.functionArn]);
+```
+
+Using the `Trail.onEvent()` API, a Cloudwatch event rule can be created that gets triggered for
+every event logged in CloudTrail.
+To only use the events that are of interest, either from a particular service, specific account or
+time range, they can be filtered down using the APIs available in `aws-events`. The following code
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-cloudtrail/README.md:100]: which APIs? is `addEventPattern` the only one or are there others?

File: packages/@aws-cdk/aws-cloudtrail/README.md [hunk 4]
```diff
@@ -92,4 +92,33 @@ trail.addLambdaEventSelector(["arn:aws:lambda"]);
...
+  code: lambda.Code.fromAsset('resource/myfunction');
+  runtime: lambda.Runtime.NODEJS_12_X,
+  handler: 'index.handler',
+});
+
+const event = Trail.onEvent(this, 'MyCloudWatchEvent', {
+  target: new eventTargets.LambdaFunction(myFunctionHandler),
+});
+
+event.addEventPattern({
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-cloudtrail/README.md:120]:
```suggestion
const rule = Trail.onEvent(this, 'MyCloudWatchEvent', {
  target: new eventTargets.LambdaFunction(myFunctionHandler),
});

rule.addEventPattern({
```

motivation: the `onEvent` API does not give us back an event, it returns a rule that we can add filtering on

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts [hunk 5]
```diff
@@ -313,14 +330,11 @@ export class Trail extends Resource {
    * be captured from any one.
    *
    * Be sure to filter the event further down using an event pattern.
+   *
+   * @deprecated - use Trail.onEvent()
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:334]: had a similar thought re: stepfunctions - how do we communicate the change of an API to deprecated. It feels like it belongs somewhere in the `CHANGELOG` but I don't know what a good mechanism is.

It's not a breaking change, but worth calling out so users can discover it outside of an IDE.

File: packages/@aws-cdk/aws-cloudtrail/README.md (hunk 3 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-cloudtrail/README.md:100]: That's the only one. I don't want to expand too much and hope the user would go to the aws-events module to learn more.

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts (hunk 5 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:334]: Ideally, the CLI on running `cdk synth` or `cdk deploy` reports a set of warnings for each deprecated API that was used.

File: packages/@aws-cdk/aws-cloudtrail/README.md (hunk 3 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cloudtrail/README.md:100]: I think that's reasonable.

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts (hunk 5 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:334]: I think that's a reasonable experience. We do have a `-strict` flag that can fail on warnings for users running in CI. Might be nice if it got a CHANGELOG mention for users who check release notes.
